<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenVpnAdvancedWidget</class>
 <widget class="QWidget" name="OpenVpnAdvancedWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>573</width>
    <height>471</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="KTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="generalTab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Gateway Port:</string>
           </property>
           <property name="buddy">
            <cstring>sbCustomPort</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="sbCustomPort">
           <property name="specialValueText">
            <string comment="like in use Automatic configuration">Automatic</string>
           </property>
           <property name="maximum">
            <number>65536</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Tunnel MTU:</string>
           </property>
           <property name="buddy">
            <cstring>sbMtu</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KIntSpinBox" name="sbMtu">
           <property name="specialValueText">
            <string comment="like in use Automatic configuration">Automatic</string>
           </property>
           <property name="maximum">
            <number>65533</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>UDP fragment size:</string>
           </property>
           <property name="buddy">
            <cstring>sbUdpFragmentSize</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KIntSpinBox" name="sbUdpFragmentSize">
           <property name="specialValueText">
            <string comment="like in use Automatic configuration">Automatic</string>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QCheckBox" name="chkUseCustomReneg">
           <property name="text">
            <string>Use custom renegotiation interval</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KIntSpinBox" name="sbCustomReneg">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="maximum">
            <number>604800</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="chkUseLZO">
         <property name="text">
          <string>Use LZO compression</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkUseTCP">
         <property name="text">
          <string>Use TCP connection</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkUseTAP">
         <property name="text">
          <string>Use TAP device</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkMssRestrict">
         <property name="text">
          <string>Restrict TCP maximum segment size (MSS)</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>306</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="securityTab">
      <attribute name="title">
       <string>Security</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Cipher:</string>
           </property>
           <property name="buddy">
            <cstring>cboCipher</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KComboBox" name="cboCipher">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <item>
            <property name="text">
             <string>Obtaining available ciphers...</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>HMAC Authentication:</string>
           </property>
           <property name="buddy">
            <cstring>cboHmac</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KComboBox" name="cboHmac">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <item>
            <property name="text">
             <string comment="like in use Default configuration">Default</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string comment="like in None setting selected">None</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>MD-4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>MD-5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>SHA-1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>SHA-224</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>SHA-256</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>SHA-384</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>SHA-512</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>RIPEMD-160</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_6">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tlsTab">
      <attribute name="title">
       <string>TLS Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Subject Match:</string>
           </property>
           <property name="buddy">
            <cstring>subjectMatch</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KLineEdit" name="subjectMatch">
           <property name="whatsThis">
            <string>Connect only to servers whose certificate matches the given subject. Example: /CN=myvpn.company.com</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="chkRemoteCertTls">
         <property name="text">
          <string>Verify peer (server) certificate usage signature</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="labelRemoteCertTls">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Remote peer certificate TLS type:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cmbRemoteCertTls">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>Server</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Client</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="useExtraTlsAuth">
         <property name="title">
          <string>Use additional TLS authentication</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="textLabel4_3">
              <property name="text">
               <string>Key:</string>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
              <property name="buddy">
               <cstring>kurlTlsAuthKey</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="KUrlRequester" name="kurlTlsAuthKey"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="textLabel1">
              <property name="text">
               <string>Key Direction:</string>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
              <property name="buddy">
               <cstring>cboDirection</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="KComboBox" name="cboDirection">
              <item>
               <property name="text">
                <string comment="like in None setting selected">None</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Server (0)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Client (1)</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_7">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>533</width>
           <height>178</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="proxyTab">
      <attribute name="title">
       <string>Proxies</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_2">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::ExpandingFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Proxy Type:</string>
         </property>
         <property name="buddy">
          <cstring>cmbProxyType</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="KComboBox" name="cmbProxyType">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <item>
          <property name="text">
           <string>Not Required</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>HTTP</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>SOCKS</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Server Address:</string>
         </property>
         <property name="buddy">
          <cstring>proxyServerAddress</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="KLineEdit" name="proxyServerAddress"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Port:</string>
         </property>
         <property name="buddy">
          <cstring>sbProxyPort</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="KIntSpinBox" name="sbProxyPort">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximum">
          <number>65535</number>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QCheckBox" name="chkProxyRetry">
         <property name="text">
          <string>Retry indefinitely when errors occur</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Proxy Username:</string>
         </property>
         <property name="buddy">
          <cstring>proxyUsername</cstring>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Proxy Password:</string>
         </property>
         <property name="buddy">
          <cstring>proxyPassword</cstring>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="KLineEdit" name="proxyUsername"/>
       </item>
       <item row="6" column="1">
        <widget class="QCheckBox" name="chkProxyShowPassword">
         <property name="text">
          <string>Show Password</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_14">
         <item>
          <widget class="KLineEdit" name="proxyPassword">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="passwordMode" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KComboBox" name="proxyPasswordStorage">
           <item>
            <property name="text">
             <string>Store</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Always Ask</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Not Required</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
  <customwidget>
   <class>KIntSpinBox</class>
   <extends>QSpinBox</extends>
   <header>knuminput.h</header>
  </customwidget>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KTabWidget</class>
   <extends>QTabWidget</extends>
   <header>ktabwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>sbCustomPort</tabstop>
  <tabstop>sbMtu</tabstop>
  <tabstop>sbUdpFragmentSize</tabstop>
  <tabstop>chkUseCustomReneg</tabstop>
  <tabstop>sbCustomReneg</tabstop>
  <tabstop>chkUseLZO</tabstop>
  <tabstop>chkUseTCP</tabstop>
  <tabstop>chkUseTAP</tabstop>
  <tabstop>chkMssRestrict</tabstop>
  <tabstop>cboCipher</tabstop>
  <tabstop>cboHmac</tabstop>
  <tabstop>subjectMatch</tabstop>
  <tabstop>chkRemoteCertTls</tabstop>
  <tabstop>cmbRemoteCertTls</tabstop>
  <tabstop>useExtraTlsAuth</tabstop>
  <tabstop>cboDirection</tabstop>
  <tabstop>cmbProxyType</tabstop>
  <tabstop>proxyServerAddress</tabstop>
  <tabstop>sbProxyPort</tabstop>
  <tabstop>chkProxyRetry</tabstop>
  <tabstop>proxyUsername</tabstop>
  <tabstop>proxyPassword</tabstop>
  <tabstop>proxyPasswordStorage</tabstop>
  <tabstop>chkProxyShowPassword</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>chkUseCustomReneg</sender>
   <signal>toggled(bool)</signal>
   <receiver>sbCustomReneg</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkRemoteCertTls</sender>
   <signal>toggled(bool)</signal>
   <receiver>labelRemoteCertTls</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>145</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkRemoteCertTls</sender>
   <signal>toggled(bool)</signal>
   <receiver>cmbRemoteCertTls</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
